File: client/src/entities/Project/model/types/Project.ts
import { Task } from '@/entities/Task';
import { Profile } from '@/entities/Profile';

export enum ProjectStatus {
    CREATED = 'created',
    PROGRESS = 'progress',
    REVIEW = 'review',
    COMPLETED = 'completed',
}

export interface Project {
    id: string;
    title: string;
    description: string;
    documents: string[];
    status: ProjectStatus;
    author: Profile;
    tasks: Task[];
    likes: {
        value: 0;
        followersId: string[];
    };
    createdAt: Date;
    startedAt: Date;
    finishedAt: Date;
}

export type SortingDirection = 'asc' | 'desc';
export type ProjectSortingKeys = 'popularity' | 'createdAt';
export type ProjectSorting = `${ProjectSortingKeys}-${SortingDirection}`;
export const sortingMapper: Record<ProjectSorting, string> = {
    'createdAt-asc': 'сначала новые',
    'createdAt-desc': 'сначала старые',
    'popularity-asc': 'сначала популярные',
    'popularity-desc': 'сначала неизвестные',
};

File: backend/server/internal/domain/types/models/project.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
	"server/internal/domain/types/dto"
	"server/internal/domain/types/enum"
	"time"
)

type Project struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Title       string             `bson:"title" json:"title"`
	Description string             `bson:"description" json:"description"`
	Status      enum.Status        `bson:"status" json:"status"`

	Likes dto.LikeData `bson:"likes" json:"likes"`

	Documents []string             `bson:"documents" json:"documents"`
	Author    primitive.ObjectID   `json:"author" bson:"author"`
	Tasks     []primitive.ObjectID `bson:"tasks" json:"tasks"`

	CreatedAt time.Time `bson:"createdAt" json:"createdAt"`
	UpdatedAt time.Time `bson:"updatedAt" json:"updatedAt"`

	StartedAt  time.Time `bson:"startedAt" json:"startedAt"`
	FinishedAt time.Time `bson:"finishedAt" json:"finishedAt"`
}
